internal/api/api.go

```diff
@@ -114,6 +117,35 @@ func (c *client) NewRequest(query string, vars map[string]interface{}) Request {
 }
 
 func (c *client) NewHTTPRequest(ctx context.Context, method, p string, body io.Reader) (*http.Request, error) {
+	if c.supportsGzip == nil {
+		version, err := c.getSourcegraphVersion(ctx)
+		if err != nil {
+			return nil, err
+		}
+		supportsGzip, err := sourcegraphVersionCheck(version, ">= 3.21.0", "2020-10-12")
+		if err != nil {
+			return nil, err
+		}
+		c.supportsGzip = &supportsGzip
+	}
+
+	if *c.supportsGzip && body != nil {
+		body = codeintelutils.Gzip(body)
+	}
+
+	req, err := c.createHTTPRequest(ctx, method, p, body)
+	if err != nil {
+		return nil, err
+	}
+
+	if *c.supportsGzip {
+		req.Header.Set("Content-Encoding", "gzip")
+	}
+
+	return req, nil
+}
```

Comment:
That means we try to gzip _every_ request — GraphQL, non-GraphQL — now, right? Doesn't that break things, since we only added gzip support to the GraphQL API?